### PR TITLE
Saving file to non-existent directory

### LINKED ISSUE #1080: Saving a file in a folder which doesn't exist doesn't create a new folder
I was trying to save a Pluto notebook in a path that doesn't exist. This is how the directory looked before:

image

Now I wanted to create a new `submission` sub-directory inside the `julia` directory and save the file there.

image

However, neither a new directory was created nor I got any errors. I think we can create the directories just like `mkdir -p` does in linux.

This feature is supported by GitHub web dashboard while adding a new file.

### PR DESCRIPTION
**Pull request by wintersm1th**
Creating path before saving the file
Close #1080

### CODE REVIEW

**Comment by pankgeorg** [comment]
Hi @wintersm1th! kudos for your first PR!

File: src/notebook/Notebook.jl [hunk 1]
```diff
@@ -120,13 +120,14 @@ end
 
 function open_safe_write(fn::Function, path, mode)
     file_content = sprint(fn)
+    mkpath(dirname(path))
```
Comment (greimel) [inline src/notebook/Notebook.jl:123]:
```suggestion
    dir = dirname(path)
    isdir(dir) || mkpath(dir)
```

**Comment by fonsp** [comment]
@greimel Isn't the purpose of `mkpath` that we don't need to do that?

**Comment by greimel** [comment]
yes, that seems right. Sorry for the noise! (I thought it behaves like `mkdir` but that it creates multiple nested folders if necessary.)

**Comment by fonsp** [comment]
@greimel No worries, thanks for contributing!

**Comment by fonsp** [comment]
@wintersm1th deleted the branch

**Review by hungpham3112 — COMMENTED** [review]
I think it should be `mkpath(path)` see #2708. If already exist `~/dir1/` but user want to create `~/dir1/dir2/dir3/` then error  appear.

**Pull request closed without merging**